packages/react-charting/src/components/SankeyChart/SankeyChart.base.tsx

```diff
@@ -152,4 +730,62 @@ export class SankeyChartBase extends React.Component<
       }
     });
   }
+  /**
+   *
+   * @param text is the text which we are trying to truncate
+   * @param rectangleWidth is the width of the rectangle which will contain the text
+   * @param padding is the space we need to leave between the rect lines and other text
+   * @param nodeWeight is the text if present needs to be accomodate in the same line as text.
+   * @returns the truncated text , if truncated given the above parameters.
+   */
+  private _truncateText(text: string, rectangleWidth: number, padding: number) {
+    const textLengthForNodeName = rectangleWidth - padding;
+    let elipsisLength = 0;
+    const tspan = select('.nodeName')
+      .append('text')
+      .attr('class', 'tempText')
+      .attr('font-size', '10')
+      .append('tspan')
+      .text(null);
+    tspan.text('...');
+    if (tspan.node() !== null) {
+      elipsisLength = tspan.node()!.getComputedTextLength();
+    }
+    tspan.text(null);
+    let line: string = '';
+    for (let i = 0; i < text.length; i++) {
+      line += text[i];
+      tspan.text(line);
+      if (tspan.node() !== null) {
+        const w = tspan.node()!.getComputedTextLength();
+        if (w > textLengthForNodeName) {
+          while (tspan.node()!.getComputedTextLength() > textLengthForNodeName - elipsisLength) {
+            line = line.slice(0, -1);
+            tspan.text(line);
+          }
+          line += '...';
+          break;
+        }
+      }
+    }
+    tspan.text(null);
+    selectAll('.tempText').remove();
+    return line;
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private _showTooltip(text: string, checkTrcuncated: boolean, evt: any) {
+    if (checkTrcuncated) {
+      const tooltip = document.getElementById('tooltip')!;
+      tooltip.innerHTML = text;
+      tooltip.style.display = 'block';
+      tooltip.style.left = evt.pageX + 'px';
+      tooltip.style.top = evt.pageY - 28 + 'px';
```

Comment:
Is there any unit tests that can be added for the sankey chart

Reply:
I will check this.